fix(auth): add error logging for sign-in

AritraDey-Dev opened the pull request:
auth): update sign-in documentation with error handling details

Updated the authentication documentation to include details on the new error logging feature for sign-in failures. Resolves issue #2301.

louh commented on client/src/dialogs/SignIn/SignInDialog.tsx:
```diff
@@ -1,6 +1,6 @@
-import React, { useState, useRef, useEffect } from 'react'
+import React, { useState, useRef, useEffect, useId } from 'react'
+// channges made here
```
Please remove this comment.

louh reviewed (changes requested):
@AritraDey-Dev thank you for this PR.
Would you mind cleaning up the branch so that the diff shows _only_ changes to the auth error logging? It's looking like some of the recent [work from this branch](https://github.com/streetmix/streetmix/pull/3180) is included as part of the diff.

This will hopefully also clear up the conflicting commit in `SignInDialog.tsx`. Thank you.

AritraDey-Dev commented:
Yes i have resolved the comment part.Now kindly verify this once.

louh commented on client/src/dialogs/Analytics/AnalyticsDialog.tsx:
```diff
@@ -151,7 +151,6 @@ function AnalyticsDialog (): React.ReactElement {
             <div className="dialog-actions">
               <CapacitySources />
               <Checkbox
-                id="show-analytics"
```
There remain a number of these files from a previous PR (https://github.com/streetmix/streetmix/pull/3180) that shouldn't be in this diff. You should be able to resolve this and clean up the PR by rebasing your changes on the current `main` branch, and it would be helpful to squash your commits to one. This would improve the diff and make it easier to review and as a record for the future. Please address this and let me know if you need any help with this. Thanks!

AritraDey-Dev replied:
sure i will solve this.

louh replied:
Hmm, this should no longer appear in the diff. Are you absolutely certain you have rebased your changes on the latest `main` branch? If so, you may need to manually remove these files from your commit.

AritraDey-Dev commented:
Rebased the logging branch onto the latest main and squashed multiple commits into one to clean up the commit history. The PR is now easier to review and contains only the relevant changes. Ready for review!

AritraDey-Dev commented:
Now i think the pr is easy to verify.

louh commented on .env.example:
```diff
@@ -20,6 +20,7 @@ AUTH0_CLIENT_SECRET=
 # APP_DOMAIN=
 # APP_PROTOCOL=
 # PORT=
+//NT9UYJ3J8FSBYKBNYJUG33NP
```
This is still in the diff.
